File: ios/Plugin/ZephyrDfuPlugin.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

// Define the plugin using the CAP_PLUGIN Macro, and
// each method the plugin supports using the CAP_PLUGIN_METHOD macro.
CAP_PLUGIN(ZephyrDfuPlugin, "ZephyrDfu",
           CAP_PLUGIN_METHOD(updateFirmware, CAPPluginReturnCallback);
           CAP_PLUGIN_METHOD(getVersion, CAPPluginReturnPromise);
)

File: ios/Plugin/ZephyrDfuPlugin.swift
import Foundation
import Capacitor

@objc(ZephyrDfuPlugin)
public class ZephyrDfuPlugin: CAPPlugin, CAPBridgedPlugin {

    public let identifier = "ZephyrDfuPlugin"
    public let jsName = "ZephyrDfu"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "updateFirmware", returnType: CAPPluginReturnCallback),
        CAPPluginMethod(name: "getVersion", returnType: CAPPluginReturnPromise)
    ]

    private let implementation = ZephyrDfu()

    @objc func updateFirmware(_ call: CAPPluginCall) {
        call.keepAlive = true
        implementation.updateFirmware(call)
    }

    @objc func getVersion(_ call: CAPPluginCall) {
        implementation.getVersion(call)
    }
}


